src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -265,7 +266,13 @@ def acs_create(resource_group_name, deployment_name, dns_name_prefix, name, ssh_
      if raw=true
     :raises: :class:`CloudError<msrestazure.azure_exceptions.CloudError>`
     """
+    if not dns_name_prefix:
+        # Use subscription id to provide uniqueness and prevent DNS name clashes
+        _, subscription_id, _ = Profile().get_login_credentials(subscription_id=None)
+        dns_name_prefix = '{}-{}'.format(name, subscription_id[0:6])
```

Comment:
I fear this doesn't protect against all name clashes. Consider these commands run in a single subscription:

```
az resource group create --name colemick-prod --location westus
az resource group create --name colemick-test --location westus
az acs create --orchestrator=kubernetes --resource-group=colemick-prod --name=kubecluster
az acs create --orchestrator=kubernetes --resource-group=colemick-test --name=kubecluster
```

Won't those result in conflicting dns prefixes, despite being a valid way of creating two clusters?